scalalib/src/test/scala/mill/scalalib/GenIdeaTests.scala

```diff
@@ -0,0 +1,71 @@
+
+  trait HelloWorldModule extends scalalib.ScalaModule {
+    def scalaVersion = "2.12.4"
+    def basePath = HelloWorldTests.workingSrcPath
+  }
+
+  object HelloWorld extends TestUtil.BaseModule with HelloWorldModule
+
+  val helloWorldEvaluator = new TestEvaluator(
+    HelloWorld,
+    outPath,
+    workingSrcPath
+  )
+
+  def tests: Tests = Tests {
+    'genIdeaTests - {
+      helloWorldEvaluator(HelloWorld.scalaVersion)
+      val x = GenIdea.xmlFileLayout(helloWorldEvaluator.evaluator, HelloWorld)
+      val pp = new scala.xml.PrettyPrinter(999, 4)
+
+      for((relPath, xml) <- GenIdea.xmlFileLayout(helloWorldEvaluator.evaluator, HelloWorld)){
+        write.over(basePath/ "generated"/ relPath, pp.format(xml))
+      }
+
+      Seq(
+        "gen-idea/idea_modules/iml" ->
+          basePath / "generated" / ".idea_modules" /".iml",
+        "gen-idea/idea_modules/root.iml" ->
+          basePath / "generated" / ".idea_modules" /"root.iml",
+        "gen-idea/idea/libraries/scala-reflect_2.12.4_scala-reflect-2.12.4-sources.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "scala-reflect_2.12.4_scala-reflect-2.12.4-sources.jar.xml",
+        "gen-idea/idea/libraries/scala-reflect_2.12.4_scala-reflect-2.12.4.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "scala-reflect_2.12.4_scala-reflect-2.12.4.jar.xml",
+        "gen-idea/idea/libraries/scala-library_2.12.4_scala-library-2.12.4.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "scala-library_2.12.4_scala-library-2.12.4.jar.xml",
+        "gen-idea/idea/libraries/modules_scala-xml_2.12_1.0.6_scala-xml_2.12-1.0.6.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "modules_scala-xml_2.12_1.0.6_scala-xml_2.12-1.0.6.jar.xml",
+        "gen-idea/idea/libraries/modules_scala-xml_2.12_1.0.6_scala-xml_2.12-1.0.6-sources.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "modules_scala-xml_2.12_1.0.6_scala-xml_2.12-1.0.6-sources.jar.xml",
+        "gen-idea/idea/libraries/scala-compiler_2.12.4_scala-compiler-2.12.4-sources.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "scala-compiler_2.12.4_scala-compiler-2.12.4-sources.jar.xml",
+        "gen-idea/idea/libraries/scala-library_2.12.4_scala-library-2.12.4-sources.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "scala-library_2.12.4_scala-library-2.12.4-sources.jar.xml",
+        "gen-idea/idea/libraries/scala-compiler_2.12.4_scala-compiler-2.12.4.jar.xml" ->
+          basePath / "generated" / ".idea" / "libraries" / "scala-compiler_2.12.4_scala-compiler-2.12.4.jar.xml",
+        "gen-idea/idea/modules.xml" ->
+          basePath / "generated" / ".idea" / "modules.xml",
+        "gen-idea/idea/misc.xml" ->
+          basePath / "generated" / ".idea" / "misc.xml",
+      ).foreach { case (resource, generated) =>
+          println("checking "+resource)
```

Comment:
It would be nice if each of these reported as a different test case. But I couldnt get nesting test working. https://github.com/lihaoyi/utest#nesting-tests

Any tips?

Reply:
nested/named tests are discovered at compile-time, so you won't be able to get this working unless you hard-code one test case for each value. Having one big test case encompassing everything is fine for now